src/citrine/informatics/constraints/scalar_range_constraint.py

```diff
@@ -15,37 +16,106 @@ class ScalarRangeConstraint(Serializable['ScalarRangeConstraint'], Constraint):
+    lower_inclusive = properties.Boolean('min_inclusive')
+    upper_inclusive = properties.Boolean('max_inclusive')
     typ = properties.String('type', default='ScalarRange')
 
     def __init__(self,
                  descriptor_key: str,
-                 max: Optional[float] = None,
+                 lower_bound: Optional[float] = None,
+                 upper_bound: Optional[float] = None,
+                 lower_inclusive: Optional[bool] = None,
+                 upper_inclusive: Optional[bool] = None,
                  min: Optional[float] = None,
-                 min_inclusive: Optional[bool] = True,
-                 max_inclusive: Optional[bool] = True,
+                 max: Optional[float] = None,
+                 min_inclusive: Optional[bool] = None,
+                 max_inclusive: Optional[bool] = None,
                  session: Optional[Session] = None):
+        if lower_bound is not None and min is not None:
+            raise ValueError("Both lower_bound and min were specified.  "
+                             "Please only specify lower_bound.")
+        if upper_bound is not None and max is not None:
+            raise ValueError("Both upper_bound and max were specified.  "
+                             "Please only specify upper_bound.")
+        if lower_inclusive is not None and min_inclusive is not None:
+            raise ValueError("Both lower_inclusive and min_inclusive were specified.  "
+                             "Please only specify lower_inclusive.")
+        if upper_inclusive is not None and max_inclusive is not None:
+            raise ValueError("Both upper_inclusive and max_inclusive were specified.  "
+                             "Please only specify upper_inclusive.")
+        if min is not None or max is not None:
+            msg = "The min/max arguments for ScalarRangeConstraint are deprecated.  " \
+                  "Please use lower_bound/upper_bound instead."
+            warn(msg, DeprecationWarning)
+        if min_inclusive is not None or max_inclusive is not None:
+            msg = "The min_inclusive/max_inclusive arguments for ScalarRangeConstraint " \
+                  "are deprecated.  " \
+                  "Please use lower_inclusive/upper_inclusive instead."
+            warn(msg, DeprecationWarning)
+
         self.descriptor_key = descriptor_key
-        self.max = max
-        self.min = min
-        self.min_inclusive = min_inclusive
-        self.max_inclusive = max_inclusive
+        self.lower_bound = lower_bound or min
+        self.upper_bound = upper_bound or max
+
+        # we have to be careful with None and boolean values
+        # None or False or True -> True, so that pattern doesn't work
+        self.lower_inclusive = True
+        if lower_inclusive is not None:
+            self.lower_inclusive = lower_inclusive
+        elif min_inclusive is not None:
+            self.lower_inclusive = min_inclusive
+
+        self.upper_inclusive = True
```

Comment:
is this line necessary?

Reply:
Yes.  This sets the default if neither upper_inclusive nor max_inclusive are set.  We could also put this in an else statement.